sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -39,6 +39,24 @@ namespace Azure { namespace Core {
    */
   class Context {
   public:
+    /**
+     * @brief A context key.
+     */
+    struct Key final
```

Comment:
Why did we decide to mark this as `final`, i.e. non-derivable?

Reply:
It would be a non-breaking change to remove `final`. And why wouln't we declare it final? It is not meant to be derived. Plus, with current code, a silicing would occur, so you'll lose all of that extra data that was added in deriving class anyway.

Reply:
Makes sense. I asked because we don't do it a lot in the SDK. Should we mark most of the classes that we don't expect folks to derive from within the SDK as `final`?

Reply:
I think we should. Plus, we can explicitly list the default generated Constructor, Destructor, Copy Constructor, Assignment, Move Constructor, move Assignment. Plus, we can put explicit first `private:` for classes.

Reply:
#2030